Add support for listing Runs in an organization

arybolovlev opened the pull request:
## Description

This PR adds support for listing Runs in an organization.

## Testing plan

1. Create a new organization.
1. Create a new workspace within the new organization.
1. Optionally to validate filters create a new agent pool and attach it to the new workspace.
1. Initialize 2 or more runs.
1. List runs for the organization. Optionally filter by workspace or agent pool name.

## External links

- [API documentation](https://developer.hashicorp.com/terraform/cloud-docs/api-docs/run#list-runs-in-an-organization)

## Output from tests

Including output from tests may require access to a TFE instance. Ignore this section if you have no environment to test against.

```
$ TFE_ADDRESS="https://app.terraform.io/" TFE_TOKEN="xLIrzcOP64xzKg.atlasv1.XXX" go test -run TestRunsListForOrganization -v

=== RUN   TestRunsListForOrganization
=== RUN   TestRunsListForOrganization/without_list_options
=== RUN   TestRunsListForOrganization/without_list_options_and_include_as_nil
=== RUN   TestRunsListForOrganization/with_list_options
=== RUN   TestRunsListForOrganization/with_workspace_included
=== RUN   TestRunsListForOrganization/without_a_valid_organization_name
=== RUN   TestRunsListForOrganization/with_filter_by_agent_pool
=== RUN   TestRunsListForOrganization/with_filter_by_workspace
--- PASS: TestRunsListForOrganization (5.76s)
    --- PASS: TestRunsListForOrganization/without_list_options (0.19s)
    --- PASS: TestRunsListForOrganization/without_list_options_and_include_as_nil (0.19s)
    --- PASS: TestRunsListForOrganization/with_list_options (0.14s)
    --- PASS: TestRunsListForOrganization/with_workspace_included (0.21s)
    --- PASS: TestRunsListForOrganization/without_a_valid_organization_name (0.00s)
    --- PASS: TestRunsListForOrganization/with_filter_by_agent_pool (0.18s)
    --- PASS: TestRunsListForOrganization/with_filter_by_workspace (0.23s)
PASS
ok      github.com/hashicorp/go-tfe     6.079s
```

brandonc commented on run.go:
```diff
@@ -21,6 +21,9 @@ type Runs interface {
 	List(ctx context.Context, workspaceID string, options *RunListOptions) (*RunList, error)
 
+	// List all the runs of the given organization.
+	ListForOrganization(ctx context.Context, organisation string, options *RunListForOrganizationOptions) (*RunList, error)
```
Minor nitpick: we use the American spelling `organization`

arybolovlev replied:
Oops! Thank you for spotting this, I fixed it everywhere!

brandonc commented on run_integration_test.go:
```diff
@@ -720,3 +720,128 @@ func TestRunCreateOptions_Marshal(t *testing.T) {
+	})
+
+	t.Run("with list options", func(t *testing.T) {
+		t.Skip("paging not supported yet in API")
```
I don't think this is true and pagination seems to work as expected. Does this test not produce the expected result?

arybolovlev replied:
You are right, I removed this `t.Skip` and this test works as expected. I updated the PR description with new test results. Thank you!

brandonc reviewed (dismissed):
This is a good, highly conventional change and I appreciate the extra effort that took. We'll be able to merge this if you were to add a CHANGELOG entry.

arybolovlev commented:
> This is a good, highly conventional change and I appreciate the extra effort that took. We'll be able to merge this if you were to add a CHANGELOG entry.

Thank you for your quick and accurate review, @brandonc! I have addressed your comments.
